src/main/java/seedu/duke/commands/ShowRateCommand.java

```diff
@@ -65,18 +67,21 @@ public String getRateString (Forex instance, BigDecimal amt) {
      *
      * @return a BigDecimal representing the numeric value
      * @throws IllegalArgumentException if the input string is non-numeric
-     * @throws InvalidNumberException   if the numeric value is negative
+     * @throws InvalidNumberException   if the numeric value is out of the range (0.01, 1bn)
      */
     public BigDecimal parseAmount () throws IllegalArgumentException, InvalidNumberException {
         String[] args = input.split(" ");
         if (args.length == 3) {
             return new BigDecimal(1);
         }
-        float val = Float.valueOf(args[3]); // Potential IllegalArgumentException
-        if (val < 0) {
-            throw new InvalidNumberException(); // For negative inputs
+        if (!args[3].matches("[0-9\\.-]+")) {
+            throw new IllegalArgumentException();
         }
-        return new BigDecimal(val);
+        BigDecimal val = new BigDecimal(args[3]); // Potential NumberFormatException
```

Comment:
Should the potential NumberFormatException as commented be unhandled? 
I suggest instead of checking with String.matches, handling the NumberFormatException accomplishes the same thing in a much clearer manner.

Reply:
Perhaps I am misunderstanding the significance of the String.matches call in relation to the issues this PR solves, please clarify if so.